### DISCUSSION TITLE
Custom Schema

### DISCUSSION THREAD

**Discussion by khurramnaseemshamsi** [TPC-Council/HammerDB#738] [General]
Hi,

By using HammerDB, Is it possible to do load test on our custom schema where our transactions heavily rely on referential constraints (FK) ?

Thank you

**Comment by sm-shaw** [comment]
Hi,
Yes of course. That is the entire purpose of HammerDB that the interface is scripted so everything can customised to any extent you wish. As an example the Oracle trace file feature builds a test entirely on the users own schema and workload https://github.com/TPC-Council/HammerDB/issues/718. Regardless of database you can write something similar for a fully bespoke test. 

However to start, if you want something generic to study the impact of foreign keys on analytic and transactional workloads look at the TPROC-H and TPROC-C workloads respectively. 

You don't mention the database you are using but the TPROC-H workloads use foreign keys extensively so you can see how they improve analytic queries by removing/modifying them etc. 

The TPROC-C workloads don't include foreign keys as this is not mandatory. However if you want to add them and do a study on the impact on OLTP performance in different databases then you can see what foreign keys to add in the TPC specification here https://www.tpc.org/TPC_Documents_Current_Versions/pdf/tpc-c_v5.11.0.pdf I'd suggest capture the timings and use the web service to view the results https://www.hammerdb.com/blog/uncategorized/hammerdb-v4-8-new-features-viewing-your-benchmark-results-with-the-hammerdb-web-service/ to see the difference between FKs enabled and disabled. This can then provide a good starting point for your study on the impact of referential integrity.
